File: src/test/java/burrows/apps/example/gif/presentation/SchedulerProviderTest.java
package burrows.apps.example.gif.presentation;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
public final class SchedulerProviderTest {
  private final SchedulerProvider sut = new SchedulerProvider();

  @Test public void testIo() {
    assertThat(sut.io()).isEqualTo(Schedulers.io());
  }

  @Test public void testUi() {
    assertThat(sut.ui()).isEqualTo(AndroidSchedulers.mainThread());
  }
}

File: src/test/kotlin/burrows/apps/example/gif/presentation/SchedulerProviderTest.kt
package burrows.apps.example.gif.presentation

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import com.google.common.truth.Truth.assertThat
import org.junit.jupiter.api.Test

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
class SchedulerProviderTest {
  private val sut = SchedulerProvider()

  @Test fun testIo() {
    assertThat(sut.io()).isEqualTo(Schedulers.io())
  }

  @Test fun testUi() {
    assertThat(sut.ui()).isEqualTo(AndroidSchedulers.mainThread())
  }
}
